guides/keras_tuner/failed_trials.py

```diff
@@ -0,0 +1,257 @@
+
+    # Define the model
+    model = keras.Sequential(
+        [
+            layers.Dense(units=units_1, input_shape=(20,)),
+            layers.Dense(units=units_2),
+            layers.Dense(units=1),
+        ]
+    )
+    model.compile(loss="mse")
+
+    # Raise an error when the model is too large
+    n_params = model.count_params()
+    if n_params > 1200:
+        # When this error is raised, it skips the retries.
+        raise keras_tuner.errors.FailedTrialError(
+            f"Model too large! It contains {n_params} params."
+        )
+    return model
+
+
+tuner = keras_tuner.GridSearch(
+    hypermodel=build_model,
+    objective="val_loss",
+    overwrite=True,
+    max_retries_per_trial=3,
+    max_consecutive_failed_trials=8,
+)
+
+# Use random data to train the model.
+tuner.search(
+    x=np.random.rand(100, 20),
+    y=np.random.rand(100, 1),
+    validation_data=(
+        np.random.rand(100, 20),
+        np.random.rand(100, 1),
+    ),
+    epochs=10,
+)
+
+# Print the results.
+tuner.results_summary()
+
+"""
+## Terminate the search programmatically
+
+When there is a bug in the code we should terminate the search immediately and
+fix the bug. You can terminate the search programmatically when your defined
+conditions are met. Raising a `FatalError` (or its subclasses `FatalValueError`,
+`FatalTypeError`, or `FatalRuntimeError`) will terminate the search regardless
+of the `max_consecutive_failed_trials` argument.
+
+Following is an example to terminate the search when the model is too large.
+"""
+
+
+def build_model(hp):
+    # Define the 2 hyperparameters for the units in dense layers
+    units_1 = hp.Int("units_1", 10, 40, step=10)
+    units_2 = hp.Int("units_2", 10, 30, step=10)
+
+    # Define the model
+    model = keras.Sequential(
+        [
+            layers.Dense(units=units_1, input_shape=(20,)),
+            layers.Dense(units=units_2),
+            layers.Dense(units=1),
+        ]
+    )
+    model.compile(loss="mse")
+
+    # Raise an error when the model is too large
+    n_params = model.count_params()
+    if n_params > 1200:
+        # When this error is raised, the search is terminated.
+        raise keras_tuner.errors.FatalError(
+            f"Model too large! It contains {n_params} params."
+        )
+    return model
+
+
+tuner = keras_tuner.GridSearch(
+    hypermodel=build_model,
+    objective="val_loss",
+    overwrite=True,
+    max_retries_per_trial=3,
+    max_consecutive_failed_trials=8,
+)
+
+try:
+    # Use random data to train the model.
+    tuner.search(
+        x=np.random.rand(100, 20),
+        y=np.random.rand(100, 1),
+        validation_data=(
+            np.random.rand(100, 20),
+            np.random.rand(100, 1),
+        ),
+        epochs=10,
+    )
+except keras_tuner.errors.FatalError:
+    print("The search is terminated.")
+
+"""
+## Takeaways
+
+In this tutorial, we learned the following things about handling failed trials
```

Comment:
"In this guide, you learn how to handle failed trials in KerasTuner:"

Reply:
Done. Why use "learn" instead of "learned" here?